python/runner/runner.py

```diff
@@ -177,63 +178,76 @@ async def run_instance(self, config, args):
         input_stream = Stream()
         asyncio.create_task(self.connect_input_stream(input_stream))
 
-        self.logger.info("Running instance...")
+        self.logger.info('Running instance...')
         result = self.sequence.run(context, input_stream, *args)
 
-        self.logger.info(f"Sending PANG")
+        self.logger.info(f'Sending PANG')
         monitoring = self.streams[CC.MONITORING]
-        pang_provides_data = {
+        pang_data = {
             'provides': '',
             'contentType': ''
         }
-        send_encoded_msg(monitoring, msg_codes.PANG, pang_provides_data)
+        send_encoded_msg(monitoring, msg_codes.PANG, pang_data)
+
+        try:
+            topic = self.sequence.topic()
+            pang_data = {
+                'requires': topic,
+                'contentType': 'text/plain'
+            }
+            send_encoded_msg(monitoring, msg_codes.PANG, pang_data)
+        except AttributeError:
+            self.logger.info('Sequence without topic, skipping sending PANG')
 
         if asyncio.iscoroutine(result):
             result = await result
         if result:
             await self.forward_output_stream(result)
         else:
-            self.logger.debug("Sequence returned no output.")
+            self.logger.debug('Sequence returned no output.')
 
         self.logger.info('Finished.')
 
 
     async def connect_input_stream(self, input_stream):
-        raw_headers = await self.streams[CC.IN].readuntil(b"\r\n\r\n")
-        header_list = raw_headers.decode().rstrip().split("\r\n")
-        headers = {
-            key.lower(): val for key, val in [el.split(": ") for el in header_list]
-        }
-        self.logger.info(f"Input headers: {repr(headers)}")
+        if (not hasattr(self.sequence, 'topic')):
+            raw_headers = await self.streams[CC.IN].readuntil(b'\r\n\r\n')
+            header_list = raw_headers.decode().rstrip().split('\r\n')
+            headers = {
+                key.lower(): val for key, val in [el.split(': ') for el in header_list]
+            }
+            self.logger.info(f'Input headers: {repr(headers)}')
+            input_type = headers['content-type']
+        else:
+            input_type = 'text/plain'
```

Comment:
I think content type should not be hardcoded.

Reply:
There is no way to get contentType from the host, I'm getting it from the sequence itself now